### DISCUSSION TITLE
JMESPath community is accepting new proposals to update the specification

### DISCUSSION THREAD

**Discussion by springcomp** [jmespath-community/jmespath.spec#16] [Announcements]
Hello JMESPath community.

<img src="https://user-images.githubusercontent.com/1768820/169634102-039f2b99-04fb-4433-b683-e2bbe07a9cc3.png" height="50" />

We are a small group of developers who are interested in taking on the mantle of maintaining this wonderful bit of technology. We have already done a significant amount of work combing through the existing repositories looking for feature requests and discussions on various topics, collecting it all under a new GitHub organization: [jmespath-community](https://github.com/jmespath-community/).

This project exists as the original implementation does not appear to be maintained and has not received any improvements for a long time. After multiple [attempts](https://github.com/jmespath/jmespath.site/issues/65) to [contact](https://github.com/jmespath/jmespath.site/issues/94) the original author, we decided to proceed further with this project.

With the exception of the language implementations, we have consolidated everything into two repositories:

[jmespath.spec](https://github.com/jmespath-community/jmespath.spec) is our repository to house all specification, testing, and discussion around the language. Previously it was being spread across the website repository and a few of the implementations.

Then there is the [refactored website](https://github.com/jmespath-community/jmespath.site), that now automatically draws all of its content from the [jmespath.spec](https://github.com/jmespath-community/jmespath.spec) repository, de-duplicating and streamlining updates to the language. We have this site hosted at https://jmespath.site/. (We would appreciate any CSS contributions too :)

Together, we would like to introduce a community effort that aims to maintain and improve JMESPath going forward.

Please [join us](https://github.com/jmespath-community/jmespath.spec/discussions) in deciding on the next iteration of the JMESPath specification.

**Comment by springcomp** [comment]
@Albibek @Derfies @GrahamCampbell @ImAbhishekTomar @Inilien @JalfResi @MadBomber @MajeedArni @Minitour @MountLion
@QAInsights @RSG6558 @RSwarnkar @Rajanusa @RichardBronosky @SCORE1387 @SDaron @SamMousa @Swyatoslaw @Thaina
@WillAbides @actliboy @adrianmace @alalinaci @amir-sabbaghi @and-semakin @apatrida @arl @arvenil @ashanbrown
@b-entangled @barlik @bjorne @bosskay972 @brlin-tw @brokedba @chadselph @chipsenkbeil @chris-armstrong @ciobis
@claylo @cs-christopher-carsey @cyriltovena @daniefer @danielaparker @danowar2k @darrenmothersele @deeplook @diazona @diehlaws

**Reply by adrianmace** [reply]
This was unsolicited.

**Reply by springcomp** [reply]
Please, forgive me for unsolicited tagging. I obviously misjudged the impact.

**Comment by springcomp** [comment]
@don-petry @douglasheldoracle @falcon9heavy @feyzee @filipenf @fjsj @frioux @gibson042 @glenveegee @gogainda
@gpacix @gpolitis @grosser @hawkeyej @impredicative @innovate-invent @iveskins @jamesls @jarrodek @jet10000
@jlamande @jose1711 @jsiembida @kbroughton @kevinkirkup @kintarowins @kloczek @kvetoslavnovak @legendetm @looker53
@marefr @mbrightcpacket @mdavis-xyz @mjpieters @mloskot @morokin @mousepotato @mtdowling @mvastola @narcoticfresh
@nebopass @nextlevelshit @np42 @ntextreme3 @olivergondza @omniproc @orsigiorgio @oylbin @philtay @pmarkert

**Comment by springcomp** [comment]
@posita @praveenambati1233 @qwordy @rathD01 @ric79 @richardgaunt @robse-git @rrjanbiah @russt @sauron918
@seitzbg @shashankmehra @shukla2112 @simonvadee @skyzyx @sliiser @spmp @springcomp @suntong @teppeis
@tmccombs @tmladek @tomelliff @twosider @vladik-hbinov @weidongcao @welkang @wryun @wungad @youngbob
@zacard-orc @zmedico

**Reply by posita** [reply]
What is this? Why am I being at-mentioned? What is required of me?

**Reply by springcomp** [reply]
Sorry for the unsolicited tagging. I certainly did not intend this to be interpreted as SPAM.
You once contributed or raised a pull request to one of the JMESPath repositories.

We thought you would be interested in the future of JMESPath that we are trying to build.

**Reply by rrjanbiah** [reply]
@springcomp Your efforts are much appreciated. I'm not very much qualified to comment or contribute anything here... except that once I suggested to offer reverse JMESPath functionality https://github.com/jmespath/jmespath.site/issues/96 and would still be interested in that (You've already answered that it is not your scope)

**Reply by posita** [reply]
What is the source for my "contribution" that lands me on this list?

**Reply by rrjanbiah** [reply]
> What is the source for my "contribution" that lands me on this list?

@posita Perhaps https://github.com/jmespath/jmespath.terminal/issues/10 & https://github.com/jmespath/jmespath.py/issues/88 (I'm not the OP)

**Comment by SamMousa** [comment]
Did you check with the authors of the original repositories? Are they abandoning their version?

Some notes should be added to the readme explaining why this community version is needed and why you chose to do it under a different organisation.

**Reply by springcomp** [reply]
@SamMousa [we tried](https://github.com/jmespath/jmespath.site/issues/65) to contact them. And [again](https://github.com/jmespath/jmespath.site/issues/94)

We have decided to proceed forward since the original maintainers seem to have abandoned the project.
Good point about adding a note explaining why we forked this effort.

As for why this version is needed, only time will tell if the community wants to have this language improved and maintained 😉.

**Reply by SamMousa** [reply]
Ah fair enough, but maybe put a mention to those issues on the readme; or something on a wiki here.

I actually remember those issues but did not recognize your handle! Sounds great!

**Reply by darrenmothersele** [reply]
My first reaction was to question the unsolicited tagging. Initial thoughts were: Is this spam? Is this endorced by the original author? Does it have permission to use the name/trademarks? Does it have wider community approval? What is the proposed govenence of the project? 

Definitely put something on the site or README.

Thanks

**Reply by springcomp** [reply]
Sorry for the unsolicited tagging. I certainly did not intend this to be interpreted as SPAM.

As we have tried repeatedly in the past to contact the original authors, we are now way past trying to seek their endorsements and decided to proceed forward. Only the community will make this effort a success or not.

We have carefully reviewed the licensing constraints and as far as we can tell:
- The `JMESPath` word itself has no trademark.
- The specification is licensed using Apache 2.0.
- The website itself needed attribution. The original website had a copyright notice but we feel we have significantly changed the website to not introduce any confusion. We expect the original website to be superceded with the new project hosted on https://jmespath.site.

We are happy to work with the community to try and iron-out anything that might be considered rude or bad etiquette on our part.

We will definitely put a notice in the README.

**Comment by skyzyx** [comment]
As a former AWS employee who worked with the creators of this project, I'm happy to reach out to them and see if I can get their attention. I too have run into numerous issues with so many things being woefully out of date.

**Reply by springcomp** [reply]
@skyzyx Please that would be awesome.

**Reply by jamesls** [reply]
Hi, I'm seeing this now.  I've been tagged in multiple places, trying to consolidate the conversation back to the original issue: https://github.com/jmespath/jmespath.site/issues/94 .

**Reply by danielaparker** [reply]
If there any options for the future of JMESPath in which James remains as the shepherd, I think those are the best options. I appreciate that a number of years have gone by and not much has happened, my own modest attempt to have a C++ implementation added to the listed implementations has languished in the pull requests for two years :-) But I have a great deal of appreciation for the work that James has done on the specifications, and trust his judgement.

**Reply by brokedba** [reply]
why has this project been abandoned by the authors . is it because more people are using jq nowadays ?
Personally I prefer JamesP

**Reply by springcomp** [reply]
I have no idea… 😏 .But it has not been abandoned by me. 😉
Indeed a lot of people use either `jq`, `JSONPath` or `Jsonata`.

Only a thriving community can make this community effort successful.
Please, support JMESPath Community !
